### PR TITLE
Indexing through leader node

### PR DESCRIPTION
**Pull request by route**

### CODE REVIEW

File: lib/solr/commit/request.rb [hunk 1]
```diff
@@ -3,9 +3,16 @@ module Commit
     class Request
       PATH = '/update'.freeze
 
-      def run
-        http_request = Solr::Request::HttpRequest.new(path: PATH, url_params: { commit: true }, method: :post)
-        Solr::Request::Runner.call(request: http_request)
+      def run(runner_options: nil)
+        http_request = Solr::Request::HttpRequest.new(path: PATH, url_params: { commit: true, openSearcher: true }, method: :post)
```
Comment (SeTeM) [inline lib/solr/commit/request.rb]: `openSearcher` default value is `true`

File: lib/solr/request/first_shard_leader_node_selection_strategy.rb [hunk 2]
```diff
@@ -25,7 +19,7 @@ def first_shard_leader_replica_node_for(collection:)
       end
 
       def solr_cloud_active_nodes_urls
```
Comment (SeTeM) [inline lib/solr/request/first_shard_leader_node_selection_strategy.rb]: shouldn't leader be excluded from this list of replicas?

File: lib/solr/commit/request.rb (hunk 1 above)
Comment (route) [reply lib/solr/commit/request.rb]: Yep there's another part I'm pushing now, it makes use of this option

File: lib/solr/request/first_shard_leader_node_selection_strategy.rb (hunk 2 above)
Comment (route) [reply lib/solr/request/first_shard_leader_node_selection_strategy.rb]: It should be but it's done with `.flatten.uniq` I didn't change much this strategy, it's refactoring

**Pull request merged**